### PR TITLE
Adaptive Card Designer - Host App for Federated Search Channel

### PR DESCRIPTION
**Pull request by naramaka**
**Creating host app for Federated Search Channel:**
The change is for new requirement to include preview card for Federate Search Channel in adaptive card designer app.
Added new card in designer template for Federated Search Channel.
Developers can use this to preview the adaptive card response on Search channel before actually deploying to prod.

Verified manually the layout . Attaching the screen shot as per current configuration.
Federated-Search-Card-Preview
Federated-Search-Channel

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/7061)

### CODE REVIEW

File: source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    this.elementsRegistry.unregister("Input.Time");
+    this.elementsRegistry.unregister("Input.Toggle");
+    this.elementsRegistry.unregister("Input.ChoiceSet");
+    this.elementsRegistry.unregister("Input.Number");
+    this.actionsRegistry.register("Action.Http", Adaptive.HttpAction);
+    this.cardHost.classList.add("fedsearch-card");
+  }
+
+  public renderTo(hostElement: HTMLElement) {
+    hostElement.classList.add("outlook-frame");
```
Comment (dclaux) [inline source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:23]: Why use an "outlook" branded frame when this HostContainer is for Federated Search? Plus this class doesn'r even seem to be defined in your css.

File: source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    hostElement.appendChild(this.cardHost);
+  }
+
+  public initialize() {
+    super.initialize();
+
+    Adaptive.GlobalSettings.useMarkdownInRadioButtonAndCheckbox = false;
+  }
+
+  private parsePadding(source: any): Adaptive.PaddingDefinition {
```
Comment (dclaux) [inline source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:33]: This seems to have been copied straight from the Outlook host container. I doubt Federated Search cards support explicit padding (and if they do they probably shouldn't). This method likely should be removed.

File: source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    return null;
+  }
+
+  public parseElement(
+    element: Adaptive.CardElement,
+    source: any,
+    context: Adaptive.SerializationContext
+  ) {
+    if (element instanceof Adaptive.Container && typeof source["rtl"] === "boolean") {
```
Comment (dclaux) [inline source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:62]: This isn't necessary. The `rtl` property has been an official AC property for a while and is parsed automatically.

File: source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+      element.rtl = source["rtl"];
+    }
+
+    if (element instanceof Adaptive.AdaptiveCard) {
```
Comment (dclaux) [inline source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:66]: All this needs to be removed... It's something that only Outlook supports.

**Review by dclaux — CHANGES_REQUESTED** [review]
Beyond my comments, the changes modify the formatting of certain existing files. Please revert to the previous formatting.

**Comment by naramaka** [comment]
Hi David ,

I had no instructions related to Federated Search changes so I had copied directly from Outlook. I have resolved the review comments.

But to your point to revert to the previous the formatting , for example I reverted the changes did to app.ts  and removed // eslint-disable-next-line no-console.
And after that it is not allowing to commit hence I had to make the changes I have made to existing files. Please find the error below:

> git -c user.useConfigOnly=true commit --quiet --allow-empty-message --file -
[STARTED] Preparing...
[SUCCESS] Preparing...
[STARTED] Running tasks...
[STARTED] Running tasks for **/!(package-lock)*
[STARTED] eslint --fix --max-warnings=0 C:/MyProjects/AdaptivecardRepo/AdaptiveCard…
[FAILED] [91meslint --fix --max-warnings=0 C:/MyProjects/AdaptivecardRepo/AdaptiveCards/source/nodejs/adaptivecards-designer-app/src/app.ts C:/MyProjects/AdaptivecardRepo/AdaptiveCards/source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts[39m [2m[FAILED][22m
[SUCCESS] Running tasks...
[STARTED] Applying modifications...
[SKIPPED] Skipped because of errors from tasks.
[STARTED] Reverting to original state because of errors...
[SUCCESS] Reverting to original state because of errors...
[STARTED] Cleaning up...
[SUCCESS] Cleaning up...

[91m[91m×[91m eslint --fix --max-warnings=0 C:/MyProjects/AdaptivecardRepo/AdaptiveCards/source/nodejs/adaptivecards-designer-app/src/app.ts C:/MyProjects/AdaptivecardRepo/AdaptiveCards/source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts:[39m

C:\MyProjects\AdaptivecardRepo\AdaptiveCards\source\nodejs\adaptivecards-designer-app\src\app.ts
  52:5  error  Unexpected console statement  no-console

✖ 1 problem (1 error, 0 warnings)

husky - pre-commit hook exited with code 1 (error)

Kindly let me know how can I resolve this.

Best Regards,
Nayana.

**Comment by dclaux** [comment]
@naramaka sorry, but I have no idea how reverting to the original formatting could possibly prevent the code from building. I'm not going to be able to help with that.

Thanks for making the code changes. However, without a new commit, I can't review your last changes.

File: source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    super(name, styleSheet);
+
+    this.actionsRegistry.unregister("Action.Submit");
+    this.elementsRegistry.unregister("Input.Text");
+    this.elementsRegistry.unregister("Input.Date");
+    this.elementsRegistry.unregister("Input.Time");
+    this.elementsRegistry.unregister("Input.Toggle");
+    this.elementsRegistry.unregister("Input.ChoiceSet");
+    this.elementsRegistry.unregister("Input.Number");
+    this.actionsRegistry.register("Action.Http", Adaptive.HttpAction);
```
[the remark this answers is not in the document]
Comment (naramaka) [reply source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts]: Its not needed. Removed it.

File: source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts (hunk 1 above)
Comment (naramaka) [reply source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts]: Removed outlook related code.

File: source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts (hunk 3 above)
Comment (naramaka) [reply source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts]: Removed.

File: source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts (hunk 4 above)
Comment (naramaka) [reply source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts]: Removed.

File: source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            action.parse(actionResources, context);
+            action.setParent(card);
+
+            actionArray.push(action);
+          }
+        });
+      }
+    }
+
+    if (element instanceof Adaptive.Image) {
```
[the remark this answers is not in the document]
Comment (naramaka) [reply source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts]: Removed

File: source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+      element.backgroundColor = source["backgroundColor"];
+    }
+
+    if (element instanceof Adaptive.Container) {
```
[the remark this answers is not in the document]
Comment (naramaka) [reply source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts]: Removed

File: source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+    if (element instanceof Adaptive.ColumnSet) {
+      const padding = this.parsePadding(source["padding"]);
+
+      if (padding) {
+        element.padding = padding;
+      }
+    }
+  }
+
+  public anchorClicked(element: Adaptive.CardElement, anchor: HTMLAnchorElement): boolean {
```
[the remark this answers is not in the document]
Comment (naramaka) [reply source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts]: Removed.

File: source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+
+    if (element instanceof Adaptive.Container) {
+      const padding = this.parsePadding(source["padding"]);
+
+      if (padding) {
+        element.padding = padding;
+      }
+    }
+
+    if (element instanceof Adaptive.ColumnSet) {
```
[the remark this answers is not in the document]
Comment (naramaka) [reply source/nodejs/adaptivecards-designer/src/containers/federated-search/federated-search-container.ts]: Removed

**Review by alok2904 — APPROVED** [review]

**Comment by dclaux** [comment]
@naramaka sorry, I had missed it (don't know how...)

**Review by alok2904 — APPROVED** [review]

**Comment by dclaux** [comment]
@naramaka just to be suere - this is the styling you're going for in search for buttons?
image

Asking because they don't necessarily look like buttons I tend to see in other places, like in Office.com for instance:
image

Also, the font used in your buttons isn't the same as the font used in the rest of the card.

**Comment by naramaka** [comment]
> @naramaka just to be suere - this is the styling you're going for in search for buttons? image
> 
> Asking because they don't necessarily look like buttons I tend to see in other places, like in Office.com for instance: image
> 
> Also, the font used in your buttons isn't the same as the font used in the rest of the card.

@dclaux There is no specification yet. I am submitting initial configuration so that we can improve it as and when we have the clarity. But if you do have any default specification then let me know I can change.

**Comment by paulcam206** [comment]
@licanhua - can you review?

**Review by licanhua — APPROVED** [review]

**Pull request merged**